### ISSUE TITLE
eslint-webpack-plugin emits no errors/warnings when using with babel-loader

### ISSUE THREAD

**Issue by mingluzhang** [webpack-contrib/eslint-webpack-plugin#59]
- Operating System: Win10
- Node Version: 12.17.0
- NPM Version: 6.14.8
- webpack Version: 4.43.0
- eslint-webpack-plugin Version: 2.4.1
- babel-loader Version: 8.1.0
- eslint-loader Version 4.0.2

### Actual Behavior

It's ok when using eslint-loader and babel-loader in webpack4 and we controlled their operation orders in config by using `enforce: "pre"` for eslint-loader. But after migrating from eslint-loader to eslint-webpack-plugin, eslint errors and warnings can't emitted any more. Suppose the problem is that we can't lint on the original source files by using eslint-webpack-plugin instead of eslint-loader.

### Code

```js
// webpack.config.js with eslint-loader
...
module.exports = env => {
    ...
    return {
        module: {
            rules: [
                {
                    test: /\.ts(x?)$/,
                    enforce: "pre",
                    include: [appPath],
                    exclude: /node_modules/,
                    loader: "eslint-loader",
                    options: {
                        fix: false,
                        emitErrors: true,
                        failOnError: true,
                    }
                },
                {
                    test: /\.ts(x?)$/,
                    include: [appPath],
                    use: [
                        {
                            loader: "babel-loader",
                            options: {
                                babelrc: false,
                                plugins: ["react-hot-loader/babel"],
                            }
                        },
                        {
                            loader: "awesome-typescript-loader",
                            options: {
                                configFileName: path.join(appPath, "tsconfig.json"),
                                silent: true,
                                useCache: true
                            }
                        }
                    ]
                }
            ]
        },
        ...
    }
}
```

```js
// webpack.config.js with eslint-webpack-plugin
const ESLintPlugin = require('eslint-webpack-plugin');
...
module.exports = env => {
    ...
    return {
        module: {
            rules: [
                {
                    test: /\.ts(x?)$/,
                    include: [appPath],
                    use: [
                        {
                            loader: "babel-loader",
                            options: {
                                babelrc: false,
                                plugins: ["react-hot-loader/babel"],
                            }
                        },
                        {
                            loader: "awesome-typescript-loader",
                            options: {
                                configFileName: path.join(appPath, "tsconfig.json"),
                                silent: true,
                                useCache: true
                            }
                        }
                    ]
                }
            ]
        },
        plugins: [
            new ESLintPlugin({
                extensions: ["ts", "tsx"],
                fix: false,
                emitError: true,
                emitWarning: true,
                failOnError: true
            }),
            ...
        ]
    }
}
```

### How Do We Reproduce?

Maybe just use eslint-webpack-plugin and babel-loader together in one project.

**Comment by netop** [comment]
For me, it appears that the build does fail if there are lint errors, it's just that they are not displayed in the console. They are also not saved in the output file, even when one is specified.

**Comment by mingluzhang** [comment]
> For me, it appears that the build does fail if there are lint errors, it's just that they are not displayed in the console. They are also not saved in the output file, even when one is specified. No TS, just regular JS/JSX.

I just receive the result of compiled successfully.

**Comment by ricardogobbosouza** [comment]
Hi @mingluzhang 
Provide a repository to reproduce
Sorry for late response

**Comment by mingluzhang** [comment]
No problem! Do you mean you need me to provide a repo which can help you to reproduce?

**Comment by ricardogobbosouza** [comment]
@mingluzhang yes, a simple repository to reproduce is of great help

**Comment by mkhattab** [comment]
@mingluzhang @ricardogobbosouza 

I'm experiencing the same issue with `eslint-webpack-plugin@2.4.1`. However, downgrading to `2.1.0`, based on the suggestion in this [other issue](https://github.com/webpack-contrib/eslint-webpack-plugin/issues/57#issuecomment-761604411), it works correctly.

**Comment by bradyclifford** [comment]
Having the same issue.

**Comment by K-Bostan** [comment]
I'm also having the same issue and it doesn't matter what webpack mode I use (development or production). Even when I'm using `webpack-dev-server`, it just hangs completely, that's the only output I get in the console:

```
ℹ ｢wds｣: Project is running at http://localhost:8080/
ℹ ｢wds｣: webpack output is served from /
ℹ ｢wds｣: Content not from webpack is served from /Users/...
ℹ ｢wdm｣: wait until bundle finished: /
✖ ｢wdm｣:
```

- Operating System: macOS Big Sur (v11.1)
- Node Version: 14.15.4
- Yarn Version: 2.4.0
- webpack Version: 5.18.0
- webpack-dev-server Version: 3.11.2
- eslint-webpack-plugin Version: 2.4.3
- eslint Version: 7.18.0
- babel-loader Version: 8.2.2
- ts-loader Version: 8.0.14
- fork-ts-checker-webpack-plugin Version: 6.1.0

```javascript
// webpack.config.ts
import * as webpack from 'webpack';
import ESLintWebpackPlugin from 'eslint-webpack-plugin';
import ForkTsCheckerWebpackPlugin from 'fork-ts-checker-webpack-plugin';
...

const configuration: webpack.Configuration = {
  ...
  module: {
    rules: [
      {
        test: /\.ts$/,
        use: [
          { loader: 'babel-loader' },
          { loader: 'ts-loader', options: { transpileOnly: true } },
        ],
      },
    ],
  },
  plugins: [
    ...
    new ESLintWebpackPlugin({
      extensions: ['js', 'ts'],
      fix: true,
      failOnError: true,
      failOnWarning: true,
    }),

    new ForkTsCheckerWebpackPlugin(),
  ],
};

```

**Comment by mingluzhang** [comment]
> @mingluzhang yes, a simple repository to reproduce is of great help
Please try [this repo](https://github.com/mingluzhang/eslint-trouble-shoot)
Also please check the problems mentioned in README.md of the repo. Thanks in advance.

**Comment by mingluzhang** [comment]
@ricardogobbosouza Is there any update on this issue?

**Comment by ricardogobbosouza** [comment]
I will see this today

**Comment by ricardogobbosouza** [comment]
Hi @mingluzhang
Very complex repository, if you can provide a simpler repository I would like to thank

**Comment by mingluzhang** [comment]
> Hi @mingluzhang
> Very complex repository, if you can provide a simpler repository

You can only care about webpack.common.config.js.
Do you mean the webpack configs are too complicated?

**Comment by mingluzhang** [comment]
> Hi @mingluzhang
> Very complex repository, if you can provide a simpler repository

https://github.com/mingluzhang/eslint-webpack-plugin-trouble-shoot
More simplified repo.

**Comment by alexander-akait** [comment]
@mingluzhang you still use `eslint-loader`, why do not use the plugin?

**Comment by mingluzhang** [comment]
> > Hi @mingluzhang
> > Very complex repository, if you can provide a simpler repository
> 
> https://github.com/mingluzhang/eslint-webpack-plugin-trouble-shoot
> More simplified repo.

@alexander-akait You can try this repo.

**Comment by nesterenko0602** [comment]
Experiencing the same problem even without `babel-loader` in my project. Made it work by downgrading to `2.1.0`

**Comment by ricardogobbosouza** [comment]
Hi @mingluzhang 

As your configuration is inside a folder, the plugin `eslint-webpack-plugin` understands that the context is `Client/Project/Configuration/` only the files inside will be linted.

To solve this, just change the `context` path:
https://github.com/mingluzhang/eslint-webpack-plugin-trouble-shoot/blob/main/Client/Project/Configuration/webpack.common.config.js#L44

```js
plugins: [new ESLintWebpackPlugin({
    context: '../', // <-- change context path
    emitError: true,
    emitWarning: true,
    failOnError: true,
    extensions: ["ts", "tsx"],
    overrideConfigFile: "../.eslintrc.js"
})],
```

**Comment by mingluzhang** [comment]
> Hi @mingluzhang
> 
> As your configuration is inside a folder, the plugin `eslint-webpack-plugin` understands that the context is `Client/Project/Configuration/` only the files inside will be linted.
> 
> To solve this, just change the `context` path:
> https://github.com/mingluzhang/eslint-webpack-plugin-trouble-shoot/blob/main/Client/Project/Configuration/webpack.common.config.js#L44
> 
> ```js
> plugins: [new ESLintWebpackPlugin({
>     context: '../', // <-- change context path
>     emitError: true,
>     emitWarning: true,
>     failOnError: true,
>     extensions: ["ts", "tsx"],
>     overrideConfigFile: "../.eslintrc.js"
> })],
> ```
> 
> See https://github.com/webpack-contrib/eslint-webpack-plugin#context

Great! Let me have a try.

**Comment by mpruser** [comment]
HI, @ricardogobbosouza
I have a question. Is the `overrideConfigFile` option a valid option?

**Comment by ricardogobbosouza** [comment]
Hi @mpruser sorry for late response
Yes, it is a valid option, all options `eslint` are valid.
See https://eslint.org/docs/developer-guide/nodejs-api#-new-eslintoptions

**Comment by ricardogobbosouza** [comment]
Its works @mingluzhang ?

**Comment by dpistoleKUBRA** [comment]
a reminder for future visitors to double check that the webpack dev server `quiet` option is not enabled; as that might manifest itself in a similar way (lint errors not showing up when running dev server, build fails silently)

not that I would ever make such a silly mistake 😊

**Comment by mqliutie** [comment]
Version : 3.0.1 has same problem..
If I install `FriendlyErrorsWebpackPlugin ` warning will be shown, otherwise warning never show unless I set `failOnWarning: true`, but I project is old, if set `true` too many place shoud be updated. 
After installing `FriendlyErrorsWebpackPlugin` webpack will compile twice

**Comment by mqliutie** [comment]
> Version : 3.0.1 has same problem..
> If I install `FriendlyErrorsWebpackPlugin ` warning will be shown, otherwise warning never show unless I set `failOnWarning: true`, but I project is old, if set `true` too many place shoud be updated.
> After installing `FriendlyErrorsWebpackPlugin` with` EslintWebpackPlugin` webpack will compile twice

My fault, In my webpack config `devServer.stats` is `errors-only`, I update to `errors-warnings` it works
